src/ledger/accountinfo.ts

```diff
@@ -34,6 +35,22 @@ export default async function getAccountInfo(
 ): Promise<FormattedGetAccountInfoResponse> {
   // 1. Validate
   validate.getAccountInfo({address, options})
+
+  // Only support retrieving account info without a tag,
+  // since account info is not distinguished by tag.
+  try {
+    const {
+      classicAddress,
+      tag
+    } = xAddressToClassicAddress(address)
+    if (tag !== false) {
+      return Promise.reject('getAccountInfo does not support the use of a tag. Use an address without a tag.')
+    }
+    address = classicAddress
+  } catch (_) {
+    // `address` is already a classic address
```

Comment:
This syntax feels awkward. 

Your error case (the `Promise.reject` should get called here. Rather than an error case being an accepted part of execution.

Reply:
This needs to be an accepted part of execution because the use of a classic address is allowed and supported.

Reply:
I would encourage you to read about why this is generally accepted as an anti pattern: https://web.archive.org/web/20140430044213/http://c2.com/cgi-bin/wiki?DontUseExceptionsForFlowControl. 

It’s relatively easy to break this into sub functions that check if it is an x-address or not.

Reply:
The reliable way to check whether it's a valid X-address is to treat it as one; if that fails, then [it is not a valid X-address](https://github.com/ripple/ripple-address-codec/blob/master/src/index.ts#L113). By the way, rippled accepts classic addresses but it also accepts other kinds of account identifiers too.

Reply:
You can still do this without using exceptions as a control flow handler though:
```javascript
let transactionRetrievalAddress = address;
if (isValidXAddress(address)) {
    transactionRetrievaAddress = decodeXAddress(xAddress);
}
```

You can even flip this on it's head and code everything to an X-address. It kinda depends on whether you'd like to represent your variables as X-addresses or classic addresses. You'd need two new functions: `isValidAddress`, `xAddressHasTag`, but both are trivial to implement. That would look like this:
```javascript
let xAddress = address;
if (isValidClassicAddress(address)) {
   xAddress = encodeXAddress(address, false)
} else if (xAddressHasTag(address)) {
   throw new Exception("Oops, we don't support tags.")
}
```

My understanding is we'd like to move towards X-Address only, so the second seems like a reasonable choice to me. You are closer to the metal on those sorts of strategic decisions in Ripple lib however, so I dee

Reply:
```
let transactionRetrievalAddress = address;
if (isValidXAddress(address)) {
    transactionRetrievaAddress = decodeXAddress(xAddress);
}
```
That's fine, but now we're decoding the address two times; `isValidXAddress` decodes the X-address in order to determine whether it's valid.

We need to have the classic address in the end because that's what rippled supports.